src/main/java/com/metamx/common/io/smoosh/FileSmoosher.java

```diff
@@ -299,38 +309,137 @@ public int bytesLeft()
     @Override
     public int write(ByteBuffer buffer) throws IOException
     {
-      WritableByteChannel channel = Channels.newChannel(out);
       return addToOffset(channel.write(buffer));
     }
 
     @Override
     public int write(InputStream in) throws IOException
     {
-      return addToOffset(Ints.checkedCast(ByteStreams.copy(in, out)));
+      return addToOffset(ByteStreams.copy(Channels.newChannel(in), channel));
```

Comment:
Agree, could be a separate PR. Also found this idiom (`ByteStream.copy(Channels.newChannel(), ...`) is used in Druid in several places, so it's more a Druid's issue than java-util's